croquemort/migrations.py

```diff
@@ -113,3 +115,34 @@ def add_hash_prefixes(self):
         self._migrate_urls_list()
         for freq in ['hourly', 'daily', 'monthly']:
             self._migrate_frequency(freq)
+
+    @rpc
+    def migrate_urls_redirect(self):
+        """
+        [migration from 1.0.0 to 2.0.0]
+
+        Migrate all the url hash fields to the new schema adopted with the
+        "redirect handling" feature.
+
+        Plus, cleanup url hashes with no url field attached.
+
+        NB: should be idempotent
+        """
+        database = self.storage.database
+        for url_hash, data in self.storage.get_all_urls():
+            if data.get('checked-url'):
+                continue
+            if data.get('url'):
+                database.hset(url_hash, 'checked-url', data['url'])
+                database.hdel(url_hash, 'url')
+                # should we really be doing that? it can be misleading
```

Comment:
> One argument for NOT filling final-url on stock data

Another one is to be able to follow the update of URLs across time. It would be interesting to know how much time it takes to re-check 10%/50%/80%/99% in prod situation (check on page or admin visit).